### DISCUSSION TITLE
My rule doens't work with function "frequency"

### DISCUSSION THREAD

**Discussion by Matheus156** [wazuh/wazuh#17177] [Q&A]
The rule id 100502 gets hits normally.

my rule:

<group name="performance\_metric,">

<rule id="100500" level="3">

<decoded_as>general_health_check</decoded_as>

<description>CPU | MEMORY | DISK usage metrics</description>

</rule>

<rule id="100502" level="11">

<if_sid>100500</if_sid>

<field name="cpu\_usage\_%%">^90|^91|^92|^93|^94|^95|^96|^97|^98|^99|^100</field>

<description>CPU usage is high: $(cpu_usage_%%)%</description>

</rule>

<rule id="100508" level="12" frequency="5" timeframe="250">

<if_matched_sid>100502</if_matched_sid>

<same_field>agent.name</same_field>

<description>CPU usage is high</description>

</rule>

</group>

**Comment by mdiego92** [comment]
Hello @Matheus156,

Did you specify the timeframe for the frequency you set up for the rule?

**Example:**

```
<rule id="3151" level="10" frequency="8" timeframe="120">
  <if_matched_sid>3102</if_matched_sid>
  <same_source_ip />
  <description>sendmail: Sender domain has bogus MX record. </description>
  <description>It should not be sending e-mail.</description>
  <mitre>
    <id>T1114</id>
    <id>T1499</id>
  </mitre>
  <group>multiple_spam,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SI.4,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
</rule>
```

**The rule is created with ID: 3151 and it will trigger a level 10 alert if the rule 3102 has matched 8 times in the last 120 seconds.**

Documentation for reference: https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html

If the issue is still occurring, could you please share with me the entire rule so I can help you with it?
Regards!
